features/ai-and-embeddings.mdx

```diff
@@ -1,22 +1,70 @@
 
-## Usage
+# Vectors
+
+### Types
+
+LibSQL uses the native SQLite BLOB [storage class](https://www.sqlite.org/datatype3.html#storage_classes_and_datatypes) for vector columns. To align with SQLite [affinity rules](https://www.sqlite.org/datatype3.html#determination_of_column_affinity), all type names have two alternatives: one that is easy to type and another with a _BLOB suffix that is consistent with affinity rules.
+
+<Info>
+We suggest library authors use type names with the `_BLOB` suffix to make results more generic and universal. For regular applications, developers can choose either alternative, as the type name only serves as a **hint** for SQLite and external extensions.
+</Info>
+
+<Info>
+As LibSQL does not introduce a new storage class, all metadata about vectors is also encoded in the `BLOB` itself. This comes at the cost of a few bytes per row but greatly simplifies the design of the feature.
+</Info>
+
+The table below lists six vector types currently supported by LibSQL. Types are listed from more precise and storage-heavy to more compact but less precise alternatives (the number of dimensions in vector $D$ is used to estimate storage requirements for a single vector).
+ 
+| Type name                  | Storage (bytes)    | Description |
+| -------------------------- | ------------------ | ----------- |
+| `FLOAT64` \| `F64_BLOB`     | $8D + 1$           | Implementation of [IEEE 754 double precision format](https://en.wikipedia.org/wiki/Double-precision_floating-point_format) for 64-bit floating point numbers |
+| `FLOAT32` \| `F32_BLOB`     | $4D$               | Implementation of [IEEE 754 single precision format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) for 32-bit floating point numbers |
+| `FLOAT16` \| `F16_BLOB`     | $2D + 1$           | Implementation of [IEEE 754-2008 half precision format](https://en.wikipedia.org/wiki/Half-precision_floating-point_format) for 16-bit floating point numbers |
+| `FLOATB16` \| `FB16_BLOB`   | $2D + 1$           | Implementation of [bfloat16 format](https://en.wikipedia.org/wiki/Bfloat16_floating-point_format) for 16-bit floating point numbers |
+| `FLOAT8` \| `F8_BLOB`       | $D + 14$           | LibSQL specific implementation which compresses each vector component to single `u8` byte `b` and reconstruct value from it using simple transformation: $\texttt{shift} + \texttt{alpha} \cdot b$ |
+| `FLOAT1BIT` \| `F1BIT_BLOB` | $\lceil \frac{D}{8} \rceil + 3$ | LibSQL-specific implementation which compresses each vector component down to 1-bit and packs multiple components into a single machine word, achieving a very compact representation |
+
+<Info>
+For most applications, the `FLOAT32` type should be a good starting point, but you may want to explore more compact options if your table has a large number of rows with vectors.
```

Comment:
Would `FLOAT16` be a better default to recommend? It's smaller but still has plenty of precision?

Reply:
Float16 can be relatively slow + it's harder to insert float16 blobs from clients (but with float32 users can just insert blobs from Float32Array)